qiskit_nature/second_q/mappers/bosonic_linear_mapper.py

```diff
@@ -0,0 +1,182 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2023.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""The Linear Mapper for Bosons."""
+
+from __future__ import annotations
+import operator
+
+from functools import reduce, lru_cache
+
+import numpy as np
+
+from qiskit.quantum_info import Pauli, SparsePauliOp
+from qiskit_nature import settings
+
+from qiskit_nature.second_q.operators import BosonicOp
+from .bosonic_mapper import BosonicMapper
+
+
+class BosonicLinearMapper(BosonicMapper):
+    """The Linear boson-to-qubit mapping.
+
+    This mapper generates a linear encoding of the Bosonic operator :math:`b_k^\\dagger, b_k` to qubit
+    operators (linear combinations of pauli strings).
+    In this linear encoding each bosonic mode is represented via :math:`n_k^{max} + 1` qubits, where
+    :math:`n_k^{max}` is the max occupation of the mode (meaning the number of states used in the
+    expansion of the mode, or equivalently the state at which the maximum excitation can take place).
+    The mode :math:`|k\\rangle` is then mapped to the occupation number vector
+    :math:`|0_{n_k^{max}}, 0_{n_k^{max} - 1},..., 0_{n_k + 1}, 1_{n_k}, 0_{n_k - 1},..., 0_{0_k}\\rangle`
+
+    It implements the formula in Section II.C of Reference [1]:
+
+    .. math::
+        b_k^\\dagger = \\sum_{n_k =0}^{n_k^{max}-1}(\\sqrt{n_k +1}\\sigma_{n_k}^{+}\\sigma_{n_k + 1}^{-})
+
+    from :math:`n_k = 0` to :math:`n_k^{max} + 1` where :math:`n_k^{max}` is the maximum occupation
+    (defined by the user).
+    In the following implementation, we explicit the operators :math:`\\sigma^+` and :math:`\\sigma^-`
+    with the Pauli matrices:
+
+    .. math::
+        \\sigma_{n_k}^+ := S_j^+ = 0.5 * (X_j + i*Y_j)\\
+        \\sigma_{n_k}^- := S_j^- = 0.5 * (X_j - i*Y_j)
+
+    The length of the qubit register is:
+    ``BosonicOp.num_modes * (BosonicLinearMapper.max_occupation + 1)``
```

Comment:
Inline code does not work well with spaces, so maybe we could make this look better if it were a single code block (might have to run black again)

```suggestion
    .. code-block:: python
    
     BosonicOp.num_modes * (BosonicLinearMapper.max_occupation + 1)
```

Reply:
Done :)